### PR TITLE
Correct unit ventilator when using zone equipment night cycle manager

### LINKED ISSUE #9635: ZoneHVAC:UnitVentilator is not reading the Availability Manager List Name correctly
Issue overview
--------------

    void GetUnitVentilatorInput

    if (!lAlphaBlanks(18)) {
        unitVent.AvailManagerListName = Alphas(18);
    }

   E+ IDD

    ZoneHVAC:UnitVentilator,
       A19, \field Availability Manager List Name
            \note Enter the name of an AvailabilityManagerAssignmentList object.
            \type object-list
            \object-list SystemAvailabilityManagerLists

also, E+ crashes with blank control zone name when CycleOnControlZone is used and err file is empty.

    AvailabilityManager:NightCycle,
      UV-NightCycleManager,
      Availability Managers - Available All Year,
      UVNCSchedule,
      CycleOnControlZone,
      0.2,
      ThermostatWithMinimumRuntime,
      300,
      ;

Suggested corrections:

UnitVentilator.cc

            if (!lAlphaBlanks(19)) {
                state.dataUnitVentilators->UnitVent(UnitVentNum).AvailManagerListName = Alphas(19);
            }

SystemAvailabilityManager.cc

                // Control zone or zonelist
                if (!lAlphaFieldBlanks(6)) {
                } else if (state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType ==
                           state.dataSystemAvailabilityManager->CycleOnControlZone) {
                    ShowSevereError(state,
                                    RoutineName + cCurrentModuleObject + "=\"" + cAlphaArgs(1) + "\" invalid " + cAlphaFieldNames(6) +
                                    "=\"" + cAlphaArgs(6) + "\" not found.");
                    ErrorsFound = true;
                }

               ** use format on above severe warning **

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added ([9635-UnitVent5Zone.idf.txt](https://github.com/NREL/EnergyPlus/files/9479043/9635-UnitVent5Zone.idf.txt))

 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes #9635
 - Unit Ventilator was reading the wrong input for availability manager list name
 - Contribution from Trane North America

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by rraustad** [comment]
Looks like this is working now. Not yet sure how to formulate a unit test to capture this.

image

**Comment by rraustad** [comment]
New warning when Control Type = CycleOnControlZone and Control Zone or Zone List Name is blank.

    ** Severe  ** GetSysAvailManagerInputs: AvailabilityManager:NightCycle = "UV-NIGHTCYCLEMANAGER" Control Zone or Zone List Name required when "Control Type" = CYCLEONCONTROLZONE.

File: src/EnergyPlus/UnitVentilator.cc [hunk 1]
```diff
@@ -578,8 +578,8 @@ namespace UnitVentilator {
                 }
             }
 
-            if (!lAlphaBlanks(18)) {
-                unitVent.AvailManagerListName = Alphas(18);
+            if (!lAlphaBlanks(19)) {
+                unitVent.AvailManagerListName = Alphas(19);
```
Comment (rraustad) [inline src/EnergyPlus/UnitVentilator.cc:582]: UV was reading the cooling coil name

File: src/EnergyPlus/SystemAvailabilityManager.cc [hunk 2]
```diff
@@ -641,10 +641,20 @@ namespace SystemAvailabilityManager {
...
+                } else if (nightCycleMgr.nightCycleControlType == NightCycleControlType::OnControlZone) {
+                    ShowSevereError(state,
+                                    format("{}{} = \"{}\" {} required when \"{}\" = {}.",
+                                           RoutineName,
+                                           cCurrentModuleObject,
+                                           cAlphaArgs(1),
+                                           cAlphaFieldNames(6),
+                                           cAlphaFieldNames(4),
+                                           cAlphaArgs(4)));
+                    ErrorsFound = true;
```
Comment (rraustad) [inline src/EnergyPlus/SystemAvailabilityManager.cc:657]: The program would crash if control zone was blank and control type = CycleOnControlZone. Program exits gracefully now.

**Comment by amirroth** [comment]
> Looks like this is working now. Not yet sure how to formulate a unit test to capture this.

I think we're a bit too militaristic about the "rule" that every PR has to have a unit test, or seemingly that every line of code needs to be covered by a unit test. I've seen some pretty funny ones recently, e.g., unit tests testing enumerations and their `string_view` arrays. This should not be a "rule", it should be a "rule of thumb". If there is something that makes sense to test as a unit test, great. If it's a simple bug fix or data structure, let's not tie ourselves into knots trying to contrive a unit test. We have more important work to do.

**Comment by mitchute** [comment]
This is a pretty obvious change. I pulled in develop and ran regression and unit tests to verify, and it all looks OK. Merging.

**Pull request merged**